third_party/terraform/tests/resource_container_cluster_test.go.erb

```diff
@@ -1648,6 +1648,40 @@ func TestAccContainerCluster_withVerticalPodAutoscaling(t *testing.T) {
 		},
 	})
 
+}
+
+func TestAccContainerCluster_withWorkloadIdentityConfig(t *testing.T) {
+	t.Parallel()
+
+	clusterName := fmt.Sprintf("cluster-test-%s", acctest.RandString(10))
+	pid := getTestProjectFromEnv()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckContainerClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerCluster_withWorkloadIdentityConfigEnabled(pid, clusterName),
+			},
+			{
+				ResourceName:        "google_container_cluster.with_workload_identity_config",
+				ImportStateIdPrefix: "us-central1-a/",
+				ImportState:         true,
+				ImportStateVerify:   true,
+			},
+			{
+				Config: testAccContainerCluster_withWorkloadIdentityConfigDisabled(clusterName),
```

Comment:
Since this test stage will recreate the cluster, it isn't testing anything novel and we can remove it.